model/qti/metadata/ontology/OntologyMetadataInjector.php

```diff
@@ -99,57 +113,116 @@ public function inject($target, array $values)
         foreach ($data as $propertyUri => $perLangData) {
             foreach (array_keys($perLangData) as $lang) {
-                $target->removePropertyValueByLg(new core_kernel_classes_Property($propertyUri), $lang);
+                $this->debug(
+                    'Remove property "%s" resource="%s" language="%s"',
+                    $propertyUri,
+                    $target->getUri(),
+                    $lang
+                );
+
+                $target->removePropertyValueByLg(
+                    new core_kernel_classes_Property($propertyUri),
+                    $lang
+                );
             }
         }
 
         // Inject new data in Ontology for target.
+        //
+        $eventManager = $this->getEventManager();
         foreach ($data as $propertyUri => $perLangData) {
             foreach ($perLangData as $lang => $d) {
                 foreach ($d as $actualData) {
-                    $target->setPropertyValueByLg(new core_kernel_classes_Property($propertyUri), $actualData[0], $lang);
+                    $this->debug(
+                        'Set property "%s" value="%s" language="%s"',
+                        $propertyUri,
+                        $actualData[0],
+                        $lang
+                    );
+
+                    $target->setPropertyValueByLg(
+                        new core_kernel_classes_Property($propertyUri),
+                        $actualData[0],
+                        $lang
+                    );
 
-                    // Send events.
-                    $eventManager = ServiceManager::getServiceManager()->get(EventManager::SERVICE_ID);
                     $metadata = $actualData[1]->getPath();
                     $metadataUri = array_pop($metadata);
-                    $eventManager->trigger(new MetadataModified($target, $metadataUri, $actualData[1]->getValue()));
+                    $eventManager->trigger(
+                        new MetadataModified($target, $metadataUri, $actualData[1]->getValue())
+                    );
                 }
             }
         }
     }
-    
+
     protected function getRuleByValue($path, $value)
     {
         $pathKey = implode('->', $path);
         $rules = $this->getInjectionRules();
-        
+
         if (isset($rules[$pathKey]) === true) {
             foreach ($rules[$pathKey] as $rule) {
                 if ($rule[1] === $value) {
                     return $rule;
                 }
             }
         }
-        
+
         return false;
     }
-    
+
     protected function getRuleByPath($path)
     {
         $pathKey = implode('->', $path);
         $rules = $this->getInjectionRules();
         if (isset($rules[$pathKey]) === true) {
             return $rules[$pathKey][0];
         }
-        
+
         return false;
     }
+
+    private function debug(string $message, ...$replacements): void
+    {
+        $this->logger->info(__CLASS__ . ': ' . vsprintf($message, $replacements));
```

Comment:
We have to take care about adding logs, cloudwatch is on of the services we pay more today in AWS. Lets try to not make it to verbose, output only in case of error or inconsistencies that could help us solve a problem. 

Check the call for this method, please :)

Reply:
I'll remove them once I finish the tests

**TODO**
- [ ] Remove logger calls in `OntologyMetadataInjector`

Reply:
Changes in this file have been reverted, so marking this thread as solved.